lib/features/email_recovery/presentation/email_recovery_controller.dart

```diff
@@ -348,6 +352,35 @@ class EmailRecoveryController extends BaseController with DateRangePickerMixin {
     popBack();
   }
 
+  String getRestorationHorizonAsString() {
+    return ((arguments!.session
+        .props[0] as Map<CapabilityIdentifier, CapabilityProperties>?)
+        ?[capabilityDeletedMessagesVault]
+        ?.props[0] as Map<String, dynamic>?)
+        ?['restorationHorizon']
+        ?? "15 days";
+  }
+
+  Duration getRestorationHorizonAsDuration() {
+    String horizonWithCorrectFormat = getRestorationHorizonAsString()
+        .replaceAll(" days", "d")
+        .replaceAll(" day", "d")
+        .replaceAll(" hours", "h")
+        .replaceAll(" hour", "h")
+        .replaceAll(" minutes", "m")
+        .replaceAll(" minute", "m")
+        .replaceAll(" seconds", "s")
+        .replaceAll(" second", "s")
+        .replaceAll(" milliseconds", "ms")
+        .replaceAll(" millisecond", "ms");
+
+    return parseDuration(horizonWithCorrectFormat, separator: ' ');
```

Comment:
We should write it in extension of Session

Reply:
done !